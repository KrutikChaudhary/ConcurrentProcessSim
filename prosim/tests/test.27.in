4 3
Proc1 6 1
DOOP 5
LOOP 5
  BLOCK 1
  DOOP 4
END
HALT

Proc2 6 0
DOOP 5
LOOP 4
  BLOCK 2
  DOOP 5
END
HALT

Proc3 6 3
DOOP 5
LOOP 5
  BLOCK 1
  DOOP 4
END
HALT

Proc4 6 2
DOOP 5
LOOP 4
  BLOCK 2
  DOOP 5
END
HALT
