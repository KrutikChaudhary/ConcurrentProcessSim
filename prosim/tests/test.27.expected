00000: process 1 new
00000: process 1 ready
00000: process 2 new
00000: process 2 ready
00000: process 3 new
00000: process 3 ready
00000: process 4 new
00000: process 4 ready
00000: process 2 running
00003: process 2 ready
00003: process 2 running
00005: process 2 blocked
00005: process 1 running
00007: process 2 ready
00007: process 1 ready
00007: process 2 running
00010: process 2 ready
00010: process 2 running
00012: process 2 blocked
00012: process 1 running
00014: process 2 ready
00014: process 1 ready
00014: process 2 running
00017: process 2 ready
00017: process 2 running
00019: process 2 blocked
00019: process 1 running
00020: process 1 blocked
00020: process 4 running
00021: process 2 ready
00021: process 1 ready
00021: process 4 ready
00021: process 2 running
00024: process 2 ready
00024: process 2 running
00026: process 2 blocked
00026: process 1 running
00028: process 2 ready
00028: process 1 ready
00028: process 2 running
00031: process 2 ready
00031: process 2 running
00033: process 2 finished
00033: process 1 running
00035: process 1 blocked
00035: process 4 running
00036: process 1 ready
00036: process 4 ready
00036: process 1 running
00039: process 1 ready
00039: process 1 running
00040: process 1 blocked
00040: process 4 running
00041: process 1 ready
00041: process 4 ready
00041: process 1 running
00044: process 1 ready
00044: process 1 running
00045: process 1 blocked
00045: process 4 running
00046: process 1 ready
00046: process 4 ready
00046: process 1 running
00049: process 1 ready
00049: process 1 running
00050: process 1 blocked
00050: process 4 running
00051: process 1 ready
00051: process 4 blocked
00051: process 1 running
00053: process 4 ready
00054: process 1 ready
00054: process 1 running
00055: process 1 finished
00055: process 4 running
00058: process 4 ready
00058: process 4 running
00060: process 4 blocked
00060: process 3 running
00062: process 4 ready
00062: process 3 ready
00062: process 4 running
00065: process 4 ready
00065: process 4 running
00067: process 4 blocked
00067: process 3 running
00069: process 4 ready
00069: process 3 ready
00069: process 4 running
00072: process 4 ready
00072: process 4 running
00074: process 4 blocked
00074: process 3 running
00075: process 3 blocked
00076: process 4 ready
00076: process 3 ready
00076: process 4 running
00079: process 4 ready
00079: process 4 running
00081: process 4 finished
00081: process 3 running
00084: process 3 ready
00084: process 3 running
00085: process 3 blocked
00086: process 3 ready
00086: process 3 running
00089: process 3 ready
00089: process 3 running
00090: process 3 blocked
00091: process 3 ready
00091: process 3 running
00094: process 3 ready
00094: process 3 running
00095: process 3 blocked
00096: process 3 ready
00096: process 3 running
00099: process 3 ready
00099: process 3 running
00100: process 3 blocked
00101: process 3 ready
00101: process 3 running
00104: process 3 ready
00104: process 3 running
00105: process 3 finished
Process 1: Run time 25, Block time 5, Wait time 25
Process 2: Run time 25, Block time 8, Wait time 0
Process 3: Run time 25, Block time 5, Wait time 75
Process 4: Run time 25, Block time 8, Wait time 48
